viewer_controls.md

```diff
@@ -0,0 +1,296 @@
+# Interaction
+
+Scope:
+Updating and synchronization of rendering controls.
+
+GUI Location:
+Main window - Right Hand Pane - Preview Tab
+Full Image Viewer - Rendering Settings Panel
+
+### General controls
+
+---
+
+**Format:**
+
+**Set-up:** ...
+
+**Control**
+- State
+  - Condition
+  	- Clarification
+- Implementation
+
+---
+
+
+**Save**
+- Disabled
+ - default 
+ - user permissions exclude action
+- Enabled
+ - user permissions allow
+   - this should be checked using ``canAnnotate`` on the image
+ - when the settings have been changed
+- At Event
+ - check against settings prior to action
+- Action
+ - saves settings for current image and current user
+   - if a lut is selected, the name of the lut will be saved e.g. ``cool.lut``
+ - trigger a "update thumbnail" event - needs to update
+   - thumbnail under User Settings
+    - thumbnail on left-hand side
+
+**Save to All**
+- Disabled 
+ - user permissions exclude action.
+- Enabled
+ - user permissions allow
+   - this should be checked using ``canAnnotate`` on the image
+- At Event
+ - check against settings prior to action
+- Action
+ - saves settings for current image and current user to all selected images or all the images
+  in the left-hand side e.g. images in dataset
+    - this method will use the ``applySettingsToSet``
+  method. see [view.py](<https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroWeb/omeroweb/webgateway/views.py>)
+ - trigger a "update thumbnail" event - needs to update
+   - thumbnail under User Settings
+    - thumbnail on left-hand side
+
+**Undo**
+- Disabled
+ - default
+- Enabled
+ - when the settings have been changed
+- At Event
+ - check against settings prior to action
+- Action
+ - undo the last rendering change
+ - trigger a "render" image event
+ - trigger a modification of the enabled flag of the Save button
+
+**Redo**
+As above except:
+- Action
+ - redo the last rendering change
+
+**Copy**
+- Enabled
+ - by default
+- Action
+ - copy the rendering settings
+
+**Paste**
+- Disabled
+ - default
+ - for the image the settings are copied from
+- Enabled
+ - when saved settings available to paste
```

Comment:
Also, copied settings must be compatible with current image - Same number of channels and same pixel type.